### DISCUSSION TITLE
SQLite boilerplate and offline data persistence

### DISCUSSION THREAD

**Discussion by fairking** [quasarframework/quasar#12460] [Ideas / Proposals]
I am trying to get sqlite working in my app and just thinking would be nice to have sqlite boilerplate in CLI in the same way as `axios` and `i18n`. So when you create a new project you can choose a `sqlite` option as well. 

I am planning to use [typeorm](https://github.com/typeorm/typeorm).
Here is a working example of it: https://github.com/jepiqueau/vue-typeorm-app

Other resources: 
https://github.com/typeorm/typeorm/issues/6854
https://github.com/nklayman/vue-cli-plugin-electron-builder/issues/699#issuecomment-609523589

I am not sure about the approach with electron, there maybe a better way by using https://capacitor-community.github.io/electron/docs/gettingstarted

Basic requirements:
- ability to use sqlite database in order to persist data locally on android/ios/desktop device,
- sqlite db file stored in file (not in browser storage, unless it is dev web browser mode),
- capacitor (android/ios) and electron (windows/linux) support,
- ...

What do you think guys?

**Comment by fairking** [comment]
No answer since then. Looks like nobody is using quasar framework for such type of applications :-(

**Reply by moksamedia** [reply]
I am looking into the same thing.

**Reply by moksamedia** [reply]
Did you get it to work?

**Reply by fairking** [reply]
> Did you get it to work?

I switched to Maui, it was easier to me to get access to the backend api. Apart of that the ui framework is Quasar in webview and there is a communication between ui and backend in place. As a .net developer it was the shortest way to me to get there. If you are not familiar with .net and rather want to use js/ts for backend, I cannot help you.

**Comment by moksamedia** [comment]
Thanks. .NET is one of the few programming worlds I've never even dipped a
toe in so I'll figure something else out.

**Reply by fairking** [reply]
Sorry mate, but writing the backend in js/ts is a nightmare to me. Have a look at Tauri if you need a strong backend but you don't want to use .net (maui). Otherwise java as the second solution.
